BlazarUI/app/scripts/components/module/BuildButton.jsx

```diff
@@ -0,0 +1,37 @@
+import React, {Component, PropTypes} from 'react';
+import {Button} from 'react-bootstrap';
+import Icon from '../shared/Icon.jsx';
+import {bindAll} from 'underscore';
+
+class BuildButton extends Component {
+
+  constructor(props) {
+    super(props);
+    bindAll(this, 'build');
+  }
+
+  build() {
+    this.props.triggerBuild();
+  }
+
+  render() {
+    if (this.props.buildTriggering) {
```

Comment:
You can still trigger a build while one is in progress, it will simply be queued for when it's done. The button is only disabled while the request is pending, which normally will be pretty fast.